streams/src/test/java/org/apache/kafka/streams/integration/AbstractJoinIntegrationTest.java

```diff
@@ -163,7 +163,7 @@ void prepareEnvironment() throws InterruptedException {
 
     @After
     public void cleanup() throws InterruptedException {
-        CLUSTER.deleteTopicsAndWait(120000, INPUT_TOPIC_LEFT, INPUT_TOPIC_RIGHT, OUTPUT_TOPIC);
+        CLUSTER.deleteAllTopicsAndWait(120000);
```

Comment:
Can you elaborate why this fixed the test?

Reply:
I'm not 100 percent sure what's the race condition here, and why it fixes the test.

Reply:
To be frank, I am not sure about the root-cause. Parameterized StreamStreamJoinIntegrationTest tests are passing for `cacheEnabled=true`  and intermittently failing for parameter `cacheEnabled=false`.

I thought we may not be cleaning some intermediate topics/change-log/? topics between parameterized runs. So I tried deleting all topics including offsets topics. With this I am not able to reproduce the failures.

I may be totally wrong.  I'll spend some more time to understand the tests and debug the issue.

Reply:
@omkreddy Were you be able to reproduce the issue locally before this PR?

Reply:
It seems we are using the same application id twice in `StreamStreamJoinIntegartionTest`
```
STREAMS_CONFIG.put(StreamsConfig.APPLICATION_ID_CONFIG, appID + "-outer");
```
This might be the root case -- deleting all topics would solve the issue, too, as it prevent to start with a corrupted state.

Reply:
I seems that `TableTableJoinIntegrationTest` has a similar issue...

Reply:
Ok..  Yes,  I have observed failures with TableTableJoinIntegrationTest too. Shall I update the  appId?

Reply:
One thing I can think of is that the changelog topic of the joining streams's materialized stores were not deleted, and hence upon starting on the next test case, the store would be pre-populated and hence cause join result to be wrong (note that the state dir is a `@Rule` which means it will get cleaned up on each test case, and in the next run it will not be the same directory path).

And the reason that it does not always fail is that we have to hit it that the two tests (with and without caching) consecutively.

So think with your fix in `2.0` and `trunk`, it is Okay to leave the appID as is for now.

Reply:
Just ran some more tests locally to confirm my suspicion.